src/store/preferences.ts

```diff
@@ -111,7 +119,7 @@ export const actions = {
           Authorization:  `Basic ${ window.btoa(`${ user }:${ password }`) }`,
           'Content-Type': 'application/x-www-form-urlencoded',
         }),
-        body: JSON.stringify(state.preferences),
+        body: JSON.stringify(payload || state.preferences),
```

Comment:
```suggestion
        body: JSON.stringify(payload ?? state.preferences),
```
Doesn't make a difference here, but seems clearer.